helix-term/src/ui/editor.rs

```diff
@@ -580,18 +580,21 @@ impl EditorView {
         }
         surface.set_string(viewport.x + 5, viewport.y, progress, base_style);
 
-        if let Some(path) = doc.relative_path() {
-            let path = path.to_string_lossy();
+        let rel_path = &doc.relative_path();
+        let path = if let Some(p) = rel_path {
+            p.to_string_lossy()
+        } else {
+            Cow::Borrowed(SCRATCH_BUFFER_NAME)
+        };
```

Comment:
```suggestion
        let rel_path = doc.relative_path();
        let path = rel_path
            .as_ref()
            .map(|p| p.to_string_lossy())
            .unwrap_or(SCRATCH_BUFFER_NAME.into());


```

Reply:
Thanks -- I could have sworn I tried all possible variations to satisfy the borrow checker, but apparently not.